### DISCUSSION TITLE
Need suggestions and clarity on couple of errors / warnings I am seeing when building the snap bundle.

### DISCUSSION THREAD

**Discussion by livingrockrises** [MetaMask/snaps#1854] [Q&A]
Can someone share more insights on below so I can read more and implement fixes. Thank you!

1. 
The snap attempted to use one or more Node.js builtins, but no browser fallback has been provided. The MetaMask Snaps CLI does not support Node.js builtins by default. If you want to use this module, you must set polyfills to true or an object with the builtins to polyfill as the key and true as the value. To disable this warning, set `stats.builtIns` to `false` in your snap config file, or add the module to the `stats.builtIns.ignore` array.

I have added ethers and few other dependencies in my snap 

2.
Failed to evaluate snap bundle in SES. This is likely due to an incompatibility with the SES environment in your snap.
Received the following error from the SES environment:
`[TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read properties of undefined (reading 'env')`

3.
Another one I've seen is : `libc++abi: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument `
not sure what this means or someone saw it earlier

**Comment by Montoya** [comment]
For 1 and 2, you likely have one or more Node JS dependencies that are not compatible with SES. Recall that Snaps is not a filesystem environment or a browser environment, so Node JS packages that expect one or the other may not work here. Here's a related doc: https://docs.metamask.io/snaps/how-to/troubleshoot/#patch-dependencies

I would try one dependency at a time and see when the Snap fails to build. I am pretty sure `ethers` works fine, curious to hear what other dependencies you have.

**Reply by Montoya** [reply]
For 2, are you trying to use `process.env`?

**Reply by Montoya** [reply]
For 1, you probably need to do some polyfills...

**Reply by FrederikBolding** [reply]
@livingrockrises Sounds like you can resolve at least some of the errors by enabling the default polyfills for whatever Node.js builtins you are missing, e.g in the snap.config.ts: 
```
polyfills: {
  buffer: true,
  process: true,
},
```

**Reply by livingrockrises** [reply]
Thank you @Montoya @FrederikBolding  this was helpful. I added polyfills for some nodejs builtins and it resolved most of the issues. 
I have added process in polyfills but I still get :` Cannot read properties of undefined (reading 'env')`

I tried one dependency at at time and found out that the[ this one  ](https://www.npmjs.com/package/@biconomy/modules) is using another local dependency which uses process.env. 

I am trying to understand more how to use patch-package and if this process issue will be solved by that. Otherwise anyway planning to write light weight version of my dependencies which works with SES.

**Reply by livingrockrises** [reply]
below one comes from the site package.

`libc++abi: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument`

Whenever I make any changes, if I remove .cache in site package above error is resolved.
